doc/README.rst

```diff
@@ -0,0 +1,255 @@
+you have to use your private GPG key, we have developed an specific script that can be
+found in section `Obtaining the password of the support account`_.
+
+The checking of the UUID is done because a template could be created from a running virtual
+machine when the support account is already created; the support account should be deleted
+before doing the snapshot, otherwise the password of the support account will be the same
+in all the images instantiated from that template.
+
+The next step, the script tries to download the userdata file from the metadata server.
+This is something that cloudinit also do, but this script can do more attempts because
+it works in the background. However, if the metadata server is not accessible
+(e.g. because the network is down) the script will finally give up. In this case, it will
+use as failback a public SSH and public GPG keys hardcoded in the images. The failback
+keys are used also if the user modify the userdata content and removes the part that
+provides the keys.
+
+Of course, the private pairs of the keys stored in the images cannot be shared among
+regions for the same reasons it is recommended that each region has its own keys.
+However a procedure could be activated to request that the support password of a
+specific VM is being encrypted using the public GPG key of the region asking for it.
+
+Finally, it must be considered that the support account can be disabled by the user,
+who has full control of the VM. For example, the support account can be deleted or
+its password changed. Also the script that create the support account can be modified
+or deleted. In that case, the administrator will lost the control over the support in
+the virtual machine.
+
+Top_
+
+
+How to generate and use the keys
+================================
+
+Software vs hardware solution
+-----------------------------
+
+This document only describes how to generate and use the keys through a software
+solution, but it is also possible use a hardware device, as an smartcard.
+The use of hardware solution is not documented here because it may be specific of
+each product, although some information will be provided AS IS to people
+interested in an introduction and some references to start with.
+
+The advantage of a hardware solution is that it provides better security. It
+also make possible a full control and audit, because with a hardware solution
+the person who uses the key cannot copy it. Most of the advantages of a
+hardware solution can be replicated using a dedicated host and an agent: support
+staff do not have access to the key and the crytographic operations are delegated
+to the agent.
+
+A hardware solution consist in a device that stores the private keys and do not
+allow to extract them. Most of these devices generate the keys by themselves
+and therefore nobody has a copy of the private key, while others allows storing
+an existing key. The people does not need to access the private key to use it,
```

Comment:
allow storing

The people does not -> It is not needed to access (or the people do not)